sdk/monitor/monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,304 @@
+   client logging](#enable-client-logging) section above.
+
+### Troubleshooting invalid Kusto query
+
+If you get an HTTP error with status code 400 (Bad Request), you may have an error in your Kusto query and you will see an error message similar to the one below.
+
+```json
+{
+  "name": "RestError",
+  "code": "BadArgumentError",
+  "statusCode": 400,
+  "request": {
+    "url": "https://api.loganalytics.io/v1/workspaces/598029db-4756-4768-87f3-d6d45ed0ebd7/query",
+    "headers": {
+      "content-type": "application/json",
+      "accept": "application/json",
+      "prefer": "REDACTED",
+      "accept-encoding": "gzip,deflate",
+      "user-agent": "azsdk-js-monitor-query/1.0.0 azsdk-js-monitor-log-query/1.0.0 core-rest-pipeline/1.7.0 Node/v14.16.1 OS/(x64-Windows_NT-10.0.19044)",
+      "x-ms-client-request-id": "6b3ee80a-c21d-4ecf-ba62-59c6ab970c83",
+      "authorization": "REDACTED",
+      "content-length": "157"
+    },
+    "method": "POST",
+    "timeout": 0,
+    "disableKeepAlive": false,
+    "streamResponseStatusCodes": {},
+    "withCredentials": false,
+    "requestId": "6b3ee80a-c21d-4ecf-ba62-59c6ab970c83",
+    "allowInsecureConnection": false,
+    "enableBrowserStreams": false
+  },
+  "details": {
+    "error": {
+      "code": "BadArgumentError",
+      "message": "The request had some invalid properties",
+      "innerError": {
+        "code": "SyntaxError",
+        "message": "A recognition error occurred in the query.",
+        "innerError": {
+          "code": "SYN0002",
+          "message": "Query could not be parsed at 'string' on line [1,4]",
+          "line": 1,
+          "pos": 4,
+          "token": "string"
+        }
+      },
+      "correlationId": "611dc354-bb4e-442f-955c-7b9469ca7f84"
+    }
+  },
+  "message": "The request had some invalid properties"
+}
+```
+
+The error message may include the line number and position where the Kusto query has an error (line 2, position 244
+in the above example). You may also refer to the [Kusto Query Language](https://docs.microsoft.com/azure/data-explorer/kusto/query) reference docs to learn more about querying logs using KQL.
+
+### Troubleshooting empty log query results
+
+If your Kusto query returns empty no logs, please validate the following:
+
+- You have the right workspace ID
+- You are setting the correct time interval for the query. Try expanding the time interval for your query to see if that
+  returns any results.
+- If your Kusto query also has a time interval, the query is evaluated for the intersection of the time interval in the
+  query string and the time interval set in the `QueryTimeInterval` param provided the query API. The intersection of
+  these time intervals may not have any logs. To avoid any confusion, it's recommended to remove any time interval in
+  the Kusto query string and use `QueryTimeInterval` explicitly.
```

Comment:
```suggestion
- If your Kusto query also has a time interval, the query is evaluated for the intersection of the time interval in the query string and the time interval set in the `QueryTimeInterval` parameter provided to the query API. The intersection of these time intervals may not have any logs. To avoid any confusion, it's recommended to remove any time interval in the Kusto query string. Use `QueryTimeInterval` explicitly.
```